src/agent/at-driver.js

```diff
@@ -70,6 +78,7 @@ export class ATDriver {
     const id = this._nextId++;
     const rawMessage = JSON.stringify({ id, ...command });
     this.log(AgentMessage.AT_DRIVER_COMMS, { direction: 'outbound', message: rawMessage });
+    if (this.hasClosed) throw new Error('AT-Driver connection unexpectedly closed');
```

Comment:
The call to `WebSocket#send` on the subsequent statement [will fail in this case](https://github.com/websockets/ws/blob/bfe1b2a623eb881a107ccf9a95042e8b22933687/lib/websocket.js#L1133-L1139). I'd prefer to remove this check and instead await a Promise built from that callback. Reason being: the "ws" library might signal failure in the "send" operation for other reasons that we're currently ignoring.